### PR TITLE
Feat/Ading label support to genesyscloud_user's routing_utilization

### PR DESCRIPTION
**Pull request by tsantos-genesys**
This is very similar to my last PR (https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/793), adding label support to agent utilization (instead of org utilization, like in that PR).

Like that PR, it calls the agent utilization api directly, and I'll clean it up once the feature is released.

---

Unrelated to this PR, go generate is not running on main, failing with:
Error executing command: unable to generate website: error rendering static website: unable to render templated website to static markdown: unable to render template "data-sources/genesyscloud_architect_datatable.md.tmpl": unable to execute template: template: docTemplate:6:18: executing "docTemplate" at <plainmarkdown>: invalid value; expected string

### CODE REVIEW

**Review by carnellj-genesys — APPROVED** [review]

File: genesyscloud/resource_genesyscloud_user.go [hunk 1]
```diff
@@ -1256,6 +1319,8 @@ func updateUserRoutingUtilization(d *schema.ResourceData, usersAPI *platformclie
 					return diag.Errorf("Failed to delete Routing Utilization for user %s: %s", d.Id(), err)
 				}
 			}
+
+			log.Printf("Updated user utilization")
```
Comment (HemanthDogiparthi12) [inline genesyscloud/resource_genesyscloud_user.go:1323]: add in a id or user to this.
this log may not be of much use for backtracking and troubleshooting

File: genesyscloud/resource_genesyscloud_user.go [hunk 2]
```diff
@@ -1231,21 +1274,41 @@ func updateUserProfileSkills(d *schema.ResourceData, usersAPI *platformclientv2.
...
+				if labelUtilizations != nil && len(labelUtilizations) > 0 {
+					routingAPI := platformclientv2.NewRoutingApiWithConfig(sdkConfig)
+					apiClient := &routingAPI.Configuration.APIClient
+
+					path := fmt.Sprintf("%s/api/v2/routing/users/%s/utilization", routingAPI.Configuration.BasePath, d.Id())
+					headerParams := buildHeaderParams(routingAPI)
+					requestPayload := make(map[string]interface{})
+					requestPayload["utilization"] = buildMediaTypeUtilizations(allSettings)
+					requestPayload["labelUtilizations"] = buildLabelUtilizationsRequest(labelUtilizations)
+					_, err = apiClient.CallAPI(path, "PUT", requestPayload, headerParams, nil, nil, "", nil)
```
Comment (HemanthDogiparthi12) [inline genesyscloud/resource_genesyscloud_user.go:1298]: does routingAPI not have direct apimethod to make this call

**Comment by charliecon** [comment]
@tsantos-genesys About the issue with `go generate`, what version of golang are you using? Maybe upgrading it could help. This project uses 1.20 

We recently upgraded "github.com/hashicorp/terraform-plugin-docs" from 0.16.0 to 0.17.0. You could test out "go generate" with a previous version of the project, before the commit e7f9422c36b6534655ab8e11c6f812bbdf8db6ab (PR #804). That would help us find out if this problem you're seeing is coming from that version upgrade

File: genesyscloud/resource_genesyscloud_user.go (hunk 2 above)
Comment (tsantos-genesys) [reply genesyscloud/resource_genesyscloud_user.go:1298]: No, as mentioned on the description and the previous PR https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/793.

The new field, labelUtilizations, is hidden in the API, which means it doesn't show up in the routingAPI, and won't show up, until the feature is released.

Once it is, I'll put up another PR removing the direct call, calling routingAPI instead.

**Comment by tsantos-genesys** [comment]
@charliecon I'm using "go version go1.21.1 darwin/arm64".

I went to commit 6c5269e, just before the one you mentioned (e7f...) and go generate works.

On commit e7f it stops working.

Idk if it's a problem with my env, just wanted to mention it

**Review by dginty4 — APPROVED** [review]

**Comment by charliecon** [comment]
> @charliecon I'm using "go version go1.21.1 darwin/arm64".
> 
> I went to commit [6c5269e](https://github.com/MyPureCloud/terraform-provider-genesyscloud/commit/6c5269e9004c58a2eddf187d71b35d91b01d1a35), just before the one you mentioned (e7f...) and go generate works.
> 
> On commit e7f it stops working.
> 
> Idk if it's a problem with my env, just wanted to mention it

I cloned your repo and saw the same error. I think it has something to do with your repository having a different name. You appended "-utilization-contract" to the end of the repo name, and the error message reads "data source entitled "genesyscloud-utilization-contract", or "genesyscloud-utilization-contract_genesyscloud_architect_datatable" does not exist" , so it is tripping up on the first page. Why the repo name is involved in the first place, I'm not sure.

**Comment by tsantos-genesys** [comment]
@charliecon thanks for figuring it out, sorry for the trouble.

**Comment by charliecon** [comment]
> @charliecon thanks for figuring it out, sorry for the trouble.

No problem. Btw when I said "Why the repo name is involved in the first place, I'm not sure." I meant that I don't know why the repo name is involved in the doc generation process. I wasn't talking about your decision to change the repo name

**Review by charliecon — APPROVED** [review]

**Pull request merged**